pkg/frontend/changefeed.go

```diff
@@ -12,23 +12,38 @@ import (
 	"github.com/Azure/ARO-RP/pkg/util/recover"
 )
 
-func (f *frontend) changefeed(ctx context.Context) {
+func (f *frontend) changefeedOcpVersions(ctx context.Context) {
 	defer recover.Panic(f.baseLog)
 
 	// f.dbOpenShiftVersions will be nil when running unit tests. Return here to avoid nil pointer panic
 	if f.dbOpenShiftVersions == nil {
 		return
 	}
 
-	frontendIterator := f.dbOpenShiftVersions.ChangeFeed()
+	ocpVersionsIterator := f.dbOpenShiftVersions.ChangeFeed()
 
 	t := time.NewTicker(10 * time.Second)
 	defer t.Stop()
 
-	f.updateFromIterator(ctx, t, frontendIterator)
+	f.updateFromIteratorOcpVersions(ctx, t, ocpVersionsIterator)
 }
 
-func (f *frontend) updateFromIterator(ctx context.Context, ticker *time.Ticker, frontendIterator cosmosdb.OpenShiftVersionDocumentIterator) {
+func (f *frontend) changefeedRoleSets(ctx context.Context) {
```

Comment:
Could you explain (or leave a comment) on the changefeed/iterator concept?

Reply:
This doc goes through it in detail: https://learn.microsoft.com/en-us/azure/cosmos-db/change-feed.

But the general idea is that rather than issuing a `DELETE` to get rid of an entry in the database, you `PUT/PATCH` it to set the `Deleting` field, and then Cosmos DB knows to remove the entry from the database after the TTL has passed. So when we are updating the role sets that are in the frontend cache, we look at each database entry's `Deleting` to see if the entry is mid-deletion, and if so, we remove it/refrain from adding it to the cache.

Is that what you were asking?

Reply:
Yep, thanks for the doc - this tracks with our current CosmosDB usage.